### PR TITLE
linter.py: Fix RESULT_SEVERITY.MINOR in comments

### PR DESCRIPTION
**Pull request by morganpdx**
Adds more information to the RESULT_SEVERITY documentation, updated with the current severity map.

Fix for https://github.com/coala/documentation/issues/25
Fix for https://github.com/coala/coala/issues/2537

### CODE REVIEW

**Comment by Udayan12167** [comment]
In one commit you add a space and in another commit you remove a space. Please remove both commits.

**Comment by Udayan12167** [comment]
in commit 9fa10c5 you need to rererence the issue you fixes with fixes or closes.

File: coalib/bearlib/abstractions/Linter.py [hunk 1]
```diff
@@ -701,7 +701,7 @@ def linter(executable: str,
 
         - ``RESULT_SEVERITY.MAJOR``: Mapped by ``error``.
```
Comment (Adrianzatreanu) [inline coalib/bearlib/abstractions/Linter.py]: hey, the commit message does not give enough info. please change the shortlog or add a better explanation to the body to show why this is needed.

File: coalib/bearlib/abstractions/Linter.py [hunk 2]
```diff
@@ -701,7 +701,7 @@ def linter(executable: str,
...
         - ``RESULT_SEVERITY.NORMAL``: Mapped by ``warning`` or ``warn``.
-        - ``RESULT_SEVERITY.INFO`: Mapped by ``info``.
+        - ``RESULT_SEVERITY.INFO`: Mapped by ``info`` .
```
Comment (Adrianzatreanu) [inline coalib/bearlib/abstractions/Linter.py]: why is this needed?

**Comment by Adrianzatreanu** [comment]
im sorry but i dont really understand whats going on with this PR, in one commit youre adding a space and in another youre removing it. it means that you could simply remove those 2 commits by cutting them off with "git rebase -i origin" so please do so :)

**Comment by Makman2** [comment]
@morganpdx you can refer to this issue:
https://github.com/coala/coala/issues/2816

**Comment by morganpdx** [comment]
Sorry I kept running up against the character limit for the commit message.  I'll go ahead and rebase, plus add all the whole severity map.

**Comment by Adrianzatreanu** [comment]
hey, this is strongly a WIP. as gitmate pointed, you have a few problems in your code that can be fixed by running "coala" over your code.

also, your rebase was done incorrectly, please ask us on gitter if you need rebase (you have a merge commit that shouldnt be there).

**Comment by morganpdx** [comment]
Please see the comment thread on the original issue - Trying to resolve the linting hurdles.

File: coalib/bearlib/abstractions/Linter.py [hunk 3]
```diff
@@ -699,10 +699,12 @@ def linter(executable: str,
         ``coalib.results.RESULT_SEVERITY`` for a result. Severity strings are
         mapped **case-insensitive**!
 
-        - ``RESULT_SEVERITY.MAJOR``: Mapped by ``error``.
-        - ``RESULT_SEVERITY.NORMAL``: Mapped by ``warning`` or ``warn``.
-        - ``RESULT_SEVERITY.MINOR``: Mapped by ``info``.
-
+        - ``RESULT_SEVERITY.MAJOR``: Mapped by ``critical`` or ``c``.
+        - ``RESULT_SEVERITY.MAJOR``: Mapped by ``fatal``, ``fail`` or ``f``.
+        - ``RESULT_SEVERITY.MAJOR``: Mapped by ``error``, ``err`` or ``e``.
```
Comment (Makman2) [inline coalib/bearlib/abstractions/Linter.py]: nope please not :D This is documentation not code :P
-->
```
- ``RESULT_SEVERITY.MAJOR``: Mapped by ``critical``, ``c``,
  ``fatal``, ``fail`` ...
```

**Comment by Makman2** [comment]
I'm writing here as this is PR relevant and not issue relevant :)

gitmate runs on every commit instead on the whole PR.  That's why still issues are raised, as your old commits did not change. You need to squash everything together. You can do this using `git rebase --interactive master` and marking every line with `fixup` instead of `pick` except the first one. For more help on that you can google or chat with us on the gitter channel, it's late right now and I don't want to explain interactive rebasing, that's quite a topic :3
But as said the community is willing to help :)

File: coalib/bearlib/abstractions/Linter.py (hunk 3 above)
Comment (morganpdx) [inline coalib/bearlib/abstractions/Linter.py]: Yeah that's what I tried to do at first, but got dinged on the line length.  I'll update again...

**Comment by morganpdx** [comment]
NP,  I understand how to squash and rebase.  I think the problem with this is the linting expectations are unclear, and maneuvering through this process is very confusing.

**Comment by Asnelchristian** [comment]
why 7 commits, it should be one, could you squashed them?
Marking WIP till then

**Comment by Makman2** [comment]
@morganpdx 
Yeah that's what newcomer issues are for, to understand the review process :)
But I think too it needs still more "quickstart-explanation", we try our best :)

**Comment by morganpdx** [comment]
This is kinda ridiculous.  I think I'm going to just close this PR and start over, this should seriously not be this difficult.

**Comment by Nosferatul** [comment]
Hi there, when you want to change the last commit use `git commit --amend` change it and push again!

**Comment by aptrishu** [comment]
@morganpdx  Hey just learn to edit your commit using `git commit --amend` and then force push it `git push -f` .
Starting over again without checking what went wrong here isn't really going to help you. :)

**Comment by sils** [comment]
@morganpdx the change isn't difficult, we're doing this to teach every newcomer about the needed git knowledge to take on bigger issues. If you don't want this, don't do it. Fixing newcomer issues isn't an issue, we can do that a lot quicker than reviewing PRs. This is meant as a service to you.

**Comment by Nosferatul** [comment]
Also you should rebase. If you are stuck in the process just ping on gitter!

**Comment by morganpdx** [comment]
@Nosferatul @aptrishu @sils thanks for the advice.  I'll hop on gitter and get some help as soon as I've got some time in the next week...work is heavy right now.  Thanks!

**Comment by Makman2** [comment]
@morganpdx if you want I can take this over for you :+1: (don't worry author information won't be lost :) )

**Comment by Makman2** [comment]
Due to inactivity, https://github.com/coala/coala/pull/2985 takes over.
